src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -0,0 +1,189 @@
+import cherrypy
+from cherrypy._cpserver import Server
+import json
+import logging
+import tempfile
+from typing import Dict, List, TYPE_CHECKING, cast, Collection
+
+from orchestrator import OrchestratorError
+from mgr_module import ServiceInfoT
+from mgr_util import verify_tls_files, build_url
+from cephadm.services.ingress import IngressSpec
+from cephadm.ssl_cert_utils import SSLCerts
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+
+
+def cherrypy_filter(record: logging.LogRecord) -> int:
+    blocked = [
+        'TLSV1_ALERT_DECRYPT_ERROR'
+    ]
+    msg = record.getMessage()
+    return not any([m for m in blocked if m in msg])
+
+
+logging.getLogger('cherrypy.error').addFilter(cherrypy_filter)
+cherrypy.log.access_log.propagate = False
+
+
+class ServiceDiscovery():
+    def __init__(self, mgr: "CephadmOrchestrator") -> None:
+        self.mgr = mgr
+        self.ssl_certs = SSLCerts()
+        self.server_port = self.mgr.service_discovery_port
+        self.server_addr = self.mgr.get_mgr_ip()
```

Comment:
if you set server_addr to '::', the server will listen on all interfaces instead of being bound only to the mgr ip - which is the same behaviour as mgr/prometheus and all the monitoring services
i.e. 
<pre>
LISTEN   0        128                    *:3000                *:*      users:(("grafana-server",pid=2740618,fd=8))     
LISTEN   0        5                      *:9283                *:*      users:(("ceph-mgr",pid=2732356,fd=40))          
LISTEN   0        128                    *:9093                *:*      users:(("alertmanager",pid=2740273,fd=8))       
LISTEN   0        128                    *:9094                *:*      users:(("alertmanager",pid=2740273,fd=3))       
LISTEN   0        128                    *:9095                *:*      users:(("prometheus",pid=2747605,fd=8))         
LISTEN   0        5                      *:9000                *:*      users:(("ceph-mgr",pid=2732356,fd=57))          
LISTEN   0        128                    *:9100                *:*      users:(("node_exporter",pid=2738740,fd=3))     
</pre>

Reply:
fixed in the last push